lib/policies/load-balancing.js

```diff
@@ -128,25 +130,21 @@ DCAwareRoundRobinPolicy.prototype.init = function (client, hosts, callback) {
   this.hosts = hosts;
   hosts.on('add', this._cleanHostCache.bind(this));
   hosts.on('remove', this._cleanHostCache.bind(this));
-  if (!this.localDc) {
-    //get the first alive local, it should be local on top
-    const hostsArray = hosts.values();
-    let dcHost;
-    for (let i = 0; i < hostsArray.length; i++) {
-      const h = hostsArray[i];
-      if (h.datacenter) {
-        this.localDc = h.datacenter;
-        dcHost = h;
-        break;
-      }
-    }
-    //this should never happen but it does not hurt
-    if (!this.localDc) {
-      return callback(new errors.DriverInternalError('Local datacenter could not be determined'));
+
+  if (client && client.options) {
+    if (this.localDc && !client.options.localDataCenter) {
+      this.client.log('warning', 'Local data center \'' + this.localDc + '\' was provided as an argument to' +
```

Comment:
I'm not sure if this should be logged at "warning" level or at "info" level: users might be using 2 execution profiles, e.g. "search" and "oltp" with 2 different dcs.

I think info level is a better match for conditional statements, like "its preferable to specify the local dc using ... _when your application is targeting a single datacenter_"